build.gradle

```diff
@@ -0,0 +1,344 @@
+    source = sourceSets.main.allJava
+    destinationDir = reporting.file(name)
+    exclude excludes["${name}"]
+    failOnError false
+  }
+
+
+  tasks.create(name: "${name}JavadocJar", type: Jar) {
+    group 'Documentation'
+    description "Build the btrace-${name} javadoc jar."
+
+    appendix name
+    classifier 'javadoc'
+    from tasks["${name}Javadoc"].getOutputs()
+  }
+}
+
+
+task buildTraces(type: Exec) {
+  group 'Build'
+  inputs.files clientJar.outputs
+  inputs.files testClasses.outputs
+  inputs.dir "src/test/traces"
+  outputs.dir "${buildDir}/classes/test/traces"
+
+  workingDir "$projectDir"
+  environment('BTRACE_HOME', "$projectDir")
+  def btracec = (osFamily == 'windows') ? 'btracec.bat' : 'btracec'
+  executable "bin/${btracec}"
+  args "-cp ${buildDir}/classes/test"
+  args "-d ${buildDir}/classes/test"
+  args fileTree(dir: "src/test/traces", include: '**/*.java')
+
+  // It would be better gradle style to do the copying in another task . . .
+  // Do we really need this?
+  doLast {
+    copy {
+      from 'src/test/traces'
+      into "${buildDir}/classes/test/traces"
+      include '**/*.java'
+    }
+  }
+}
+
+
+def distContent = copySpec {
+  into('bin') {
+    from 'bin'
+  }
+
+  into('build') {
+    from "$buildDir"
+    include '*.jar'
+  }
+
+  into ('docs') {
+    from 'docs'
+    include 'javadoc/**'
+    include 'usersguide.html'
+  }
+
+  into ('samples') {
+    from 'samples'
+  }
+
+  into('.') {
+    from '.'
+    include '*.md'
+    include '*.txt'
+    include 'COPYRIGHT'
+    include 'LICENSE'
+  }
+}
+
+task createBinDistZip(type: Zip) {
+  group 'Build'
+  description 'Build the binary distribution zip-file.'
+  inputs.files agentJar.outputs
+  inputs.files bootJar.outputs
+  inputs.files clientJar.outputs
+
+  appendix = 'bin'
+  with distContent
+}
+assemble.dependsOn createBinDistZip
+
+task createBinDistTgz(type: Tar) {
+  group 'Build'
+  description 'Build the binary distribution tgz-file.'
+  inputs.files agentJar.outputs
+  inputs.files bootJar.outputs
+  inputs.files clientJar.outputs
+  compression = Compression.GZIP
+
+  appendix = 'bin'
+  with distContent
+}
+assemble.dependsOn createBinDistTgz
+
+
+ospackage {
+  maintainer = 'Jaroslav Bachorik'
+
+  if (project.version.endsWith('-SNAPSHOT')) {
+    version = project.version - '-SNAPSHOT'
+    release = (new Date()).format('YYYYMMdd.HHmmss')
+  }
+
+  postInstall file('packaging/debian/btrace/DEBIAN/postinst')
+
+  requires 'openjdk-7-jdk'
+
+  from(buildDir) {
+    into '/usr/lib/btrace'
+    include 'btrace-*.jar'
+  }
+
+  from('packaging/debian/btrace/usr/bin') {
+    into '/usr/bin'
+  }
+}
+
+buildDeb {
+  inputs.files agentJar.outputs
+  inputs.files bootJar.outputs
+  inputs.files clientJar.outputs
+}
+assemble.dependsOn buildDeb
+
+buildRpm {
+  inputs.files agentJar.outputs
+  inputs.files bootJar.outputs
+  inputs.files clientJar.outputs
+}
+assemble.dependsOn buildRpm
+
+
+test {
+  inputs.files buildTraces.outputs
+}
+
```

Comment:
This would make the build more non-standard and would require additional changes for travis-ci.
Imho this is a high price for saving little time: On my box assembling the distributions took 637ms (which we possibly could save) compared to 12.1s overall build time (7.6s for test).